### PR TITLE
LibCore: Refactor EventLoop and Object to use TimerId

### PR DESCRIPTION
**Pull request by idispatch**
- Addressed clang-tidy warnings in EventLoop and Object
- Use const as per clang-tidy
- Disambiguate SignalNumber and HandlerId in SignalHandlers

### CODE REVIEW

File: Userland/Libraries/LibCore/Event.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ class DeferredInvocationEvent : public Event {
 
 public:
     DeferredInvocationEvent(NonnullRefPtr<DeferredInvocationContext> context, Function<void()> invokee)
-        : Event(Event::Type::DeferredInvoke)
+        : Event(Event::DeferredInvoke)
```
Comment (linusg) [inline Userland/Libraries/LibCore/Event.h:56]: `Event::Type` should be an enum class instead, make that change instead if you want to improve this

File: Userland/Libraries/LibCore/EventLoop.cpp [hunk 2]
```diff
@@ -211,21 +214,21 @@ class InspectorServerConnection : public Object {
...
-        auto serialized = response.to_string();
+        auto const serialized = response.to_string();
         auto bytes_to_send = serialized.bytes();
         u32 length = bytes_to_send.size();
         // FIXME: Propagate errors
         auto sent = MUST(m_socket->write({ (u8 const*)&length, sizeof(length) }));
         VERIFY(sent == sizeof(length));
         while (!bytes_to_send.is_empty()) {
-            size_t bytes_sent = MUST(m_socket->write(bytes_to_send));
+            const size_t bytes_sent = MUST(m_socket->write(bytes_to_send));
```
Comment (linusg) [inline Userland/Libraries/LibCore/EventLoop.cpp:224]: east const as per https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#const-placement

File: Userland/Libraries/LibCore/EventLoop.h [hunk 3]
```diff
@@ -62,8 +62,10 @@ class EventLoop {
 
     bool was_exit_requested() const { return m_exit_requested; }
 
-    static int register_timer(Object&, int milliseconds, bool should_reload, TimerShouldFireWhenNotVisible);
-    static bool unregister_timer(int timer_id);
+    using TimerId = int;
+    static constexpr TimerId kInvalidTimerId = 0;
```
Comment (linusg) [inline Userland/Libraries/LibCore/EventLoop.h:66]: We don't use this naming style, please adhere to https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#names

**Review by linusg — CHANGES_REQUESTED** [review]
This needs to be split up into multiple atomic comments instead of mashing a bunch of unrelated things into one.

Also: https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#code-submission-policy

> Don't: Use weasel-words like "refactor" or "fix" to avoid explaining what's being changed.

File: Userland/Libraries/LibCore/Event.h [hunk 4]
```diff
@@ -66,17 +66,18 @@ class DeferredInvocationEvent : public Event {
 
 class TimerEvent final : public Event {
 public:
-    explicit TimerEvent(int timer_id)
+    using TimerId = int;
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/Event.h:69]: These new IDs you've created would be better as `AK_TYPEDEF_DISTINCT_ORDERED_ID`s. from AK/DistinctNumeric.h. That way they cannot alias or convert with each other.

File: Userland/Libraries/LibCore/Object.h [hunk 5]
```diff
@@ -205,9 +205,11 @@ class Object
     virtual void did_end_inspection() { }
 
 private:
+    using TimerId = int;
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/Object.h:208]: If both Core::Object and Core::EventLoop have the same concept for TimerID, why not use the same exact type for each instance? i.e. Core::EventLoop should be using a DistinctNumeric defined with the macro I mentioned above in Object.h. e.g. Core::TimerId.

**Pull request closed without merging**